install/abstract-activate.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @package Polylang
+ *
+ * /!\ THE CONSTANT `POLYLANG_BASENAME` MUST BE DEFINED.
+ */
+
+/**
+ * A generic activation class compatible with multisite.
+ *
+ * @since 3.8
+ */
+abstract class PLL_Abstract_Activate extends PLL_Abstract_Activable {
+	/**
+	 * Adds the required hooks.
+	 *
+	 * @since 3.8
+	 *
+	 * @return void
+	 */
+	public static function add_hooks(): void {
+		// Plugin activation.
+		register_activation_hook( pll_get_constant( 'POLYLANG_BASENAME', '' ), array( static::class, 'do_for_all_blogs' ) );
```

Comment:
Thinking to re-usability, what do you think to introduce a method `get_plugin_basename()`? This one could be in `PLL_Abstract_Activable`.
For another plugin, we could re-use the class just overriding this method instead of rewriting `add_hooks()`.